classes/class.ilObjMumieTaskGUI.php

```diff
@@ -469,18 +473,36 @@ public function initLPSettingsForm()
      */
     public function submitLPSettings()
     {
-        $this->initLPSettingsForm();
+        $this->plugin->includeClass('class.ilMumieTaskLPStatus.php');
+        $disable_grade_pool_selection = $this->object->isGradepoolSet();
+        $this->initLPSettingsForm($disable_grade_pool_selection);
         if (!$this->form->checkInput()) {
             $this->form->setValuesByPost();
             $this->tpl->setContent($this->form->getHTML());
             return;
         }
         $force_grade_update = $this->object->getPassingGrade() !== $this->form->getInput('passing_grade');
+        $is_gradepool_setting_update = $this->object->getPrivateGradepool() !== $this->form->getInput('privategradepool');
         $this->object->setLpModus($this->form->getInput('lp_modus'));
+        $this->object->setPrivateGradepool($this->form->getInput('privategradepool'));
         $this->object->setPassingGrade($this->form->getInput('passing_grade'));
         $this->object->doUpdate();
+
+        global $DIC;
+        $tree = $DIC['tree'];
+        $parent_ref = $tree->getParentId($this->object->getRefId());
+
+        if (true)#$is_gradepool_setting_update)
```

Comment:
You we're not supposed to delete that if clause but to update the condition ;)

We're declaring a variable `$is_gradepool_setting_update` - I suggest using that instead of `if (true)`